e2e/localnet/network/node.go

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: MIT
+//
+// Copyright (c) 2023 Berachain Foundation
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+package localnet
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+
+	"pkg.berachain.dev/polaris/e2e/localnet/container"
+)
+
+const (
+	defaultTimeout            = 30 * time.Second
+	nodeStartTime             = 10 * time.Second
+	initialBlockHeight uint64 = 2
+)
+
+// ContainerizedNode is an interface for a containerized network.
+type ContainerizedNode interface {
+	Start() error
+	Stop() error
+	Reset() error
+	Remove() error
+	GetHTTPAddress() string
+	GetWSAddress() string
+	EthClient() *ethclient.Client
+	EthWsClient() *ethclient.Client
+	WaitForBlock(number uint64) error
+}
+
+// containerizedNode implements ContainerizedNode.
+type containerizedNode struct {
+	containerClient container.Client
+	httpAddress     string
+	wsAddress       string
+	ethClient       *ethclient.Client
+	ethWsClient     *ethclient.Client
+}
+
+// NewContainerizedNode creates an implementation of Localnet using Docker. The node will be past
+// block 2 by the time node is available.
```

Comment:
this seems like a ghetto asssumption

Reply:
whats a good block number to start from. I assumed we actually want to use the chain after its on block 1/2

Reply:
I don't think we make an assumption

Reply:
its not an assumption. im just waiting to return until current block height is 2

Reply:
eh the way its worded is kinda weird. Personally it should hang until either a) a timeout b) any block is detected, irrespective of number

Reply:
fair, thats more clear